e2e2/test/cases/netpol/main_test.go

```diff
@@ -0,0 +1,333 @@
+			if err != nil {
+				log.Print("Failed to install latest aws-vpc-cni on cluster")
+				return ctx, err
+			}
+
+			// 2. Create three namespaces
+			for _, ns := range namespaces {
+				err = createNamespace(ns, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create namespace %s", ns)
+					return ctx, err
+				}
+			}
+
+			// 3. Create deployment and service
+			for ns, server := range servers {
+				err = createServerAndService(ns, server, 1, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create deployment and service for %s", server)
+					return ctx, err
+				}
+			}
+
+			return ctx, nil
+		},
+	)
+
+	testenv.Finish(
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+
+			if err != nil {
+				return ctx, err
+			}
+
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "a", Namespace: "a"}})
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "b", Namespace: "b"}})
+			client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "c", Namespace: "c"}})
+
+			return ctx, nil
+		},
+	)
+
+	os.Exit(testenv.Run(m))
+}
+
+func installLatestCNI(config *envconf.Config) error {
+
+	repoName := "eks"
+	repoUrl := "https://aws.github.io/eks-charts"
+	chartName := "eks/aws-vpc-cni"
+	releaseName := "aws-vpc-cni"
+	releaseNamespace := "kube-system"
+
+	resources := []string{"daemonset/aws-node", "clusterRole/aws-node", "clusterRoleBinding/aws-node", "serviceAccount/aws-node", "configMap/amazon-vpc-cni"}
+
+	for _, resource := range resources {
+		if _, err := fwext.ResourceExists(resource, releaseNamespace, config.KubeconfigFile()); err == nil {
+			log.Printf("Patching resource: %s labels and annotations..", resource)
+			fwext.Label(resource, "app.kubernetes.io/managed-by=Helm", releaseNamespace, config.KubeconfigFile())
+			fwext.Annotate(resource, fmt.Sprintf("meta.helm.sh/release-name=%s", releaseName), releaseNamespace, config.KubeconfigFile())
+			fwext.Annotate(resource, fmt.Sprintf("meta.helm.sh/release-namespace=%s", releaseNamespace), releaseNamespace, config.KubeconfigFile())
+		} else {
+			log.Printf("Resource %s does not exist, skipping patching labels/annotations..", resource)
+		}
+	}
+
+	values := map[string]interface{}{
+		"enableNetworkPolicy": "true",
+		"originalMatchLabels": "true",
+	}
+
+	log.Printf("Installing helm chart %s", chartName)
+	_, err := fwext.InstallPackagedRelease(repoName, repoUrl, chartName, releaseName, releaseNamespace, values)
+	if err != nil {
+		return err
+	}
+
+	cniDS := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: releaseNamespace},
+	}
+
+	wait.For(conditions.New(config.Client().Resources()).ResourceMatch(cniDS, func(object k8s.Object) bool {
+		d := object.(*appsv1.DaemonSet)
+		status := false
+		if d.Status.DesiredNumberScheduled == d.Status.NumberReady {
```

Comment:
Can you use the existing utility in the framework for your daemonset checks?

https://github.com/kubernetes-sigs/e2e-framework/blob/c61338a3773c2b64550d0e46037bfe541fa54645/klient/wait/conditions/conditions.go#L314

Reply:
Seems like we need a e2e-framework release to use this utility. I've added a TODO to update to use this when we have a new version of the framework

Reply:
we could pull in the latest changes with a go `v0.0.0-$COMMIT` version, but you could also use the one I've put in the frameworkextensions package: https://github.com/aws/aws-k8s-tester/blob/main/e2e2/internal/framework_extensions/conditions.go#L32